lib/matplotlib/backends/backend_wx.py

```diff
@@ -1144,6 +1147,31 @@ def __init__(self, canvas):
     zoomStartX = cbook._deprecate_privatize_attribute("3.3")
     zoomStartY = cbook._deprecate_privatize_attribute("3.3")
 
+    @staticmethod
+    def _icon(name):
+        """
+        Construct a `wx.Bitmap` suitable for use as icon from an image file
+        *name*, including the extension and relative to Matplotlib's "images"
+        data directory.
+        """
+        image = np.array(PIL.Image.open(cbook._get_data_path("images", name)))
+        try:
+            dark = wx.SystemSettings.GetAppearance().IsDark()
+        except AttributeError:  # wxpython < 4.1
+            # copied from wx's IsUsingDarkBackground / GetLuminance.
+            bg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_WINDOW)
+            fg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_WINDOWTEXT)
+            # See wx.Colour.GetLuminance.
+            bg_lum = (.299 * bg.red + .587 * bg.green + .114 * bg.blue) / 255
+            fg_lum = (.299 * fg.red + .587 * fg.green + .114 * fg.blue) / 255
+            dark = fg_lum - bg_lum > .2
```

Comment:
Why calculate this differently from GTK?

Reply:
This is copypasted from the implementation of wx's IsUsingDarkBackground (which is not available on older wx), so at least it's internally consistent across wx versions.

Also I was not able to figure out how to get the text color on gtk...

Reply:
I think there's a better way to do it on GTK, but it would be a much more substantial re-write.

Reply:
I'm not actually a gtk expert at all (tbh I find its docs much harder to follow than wx, which I'm not that familiar with either (I (used to) use qt fairly regularly) but find much easier to follow), so I'll leave it up to you whether we want the "complicated" solution here.

Reply:
OK, I think I've figured it out and it's actually not as complicated as I thought, but it's a bit of a hack right now. I'll try and figure out how to be less hacky and open a PR.